index.js

```diff
@@ -184,20 +184,125 @@ class Client {
+
+        if (redirectAllRequestsTo.protocol) {
+          confirmString(redirectAllRequestsTo.protocol, 'hostName must be a string');
+          confirmInArray(
+            redirectAllRequestsTo.protocol.toLowerCase(), 
+            ['http', 'https'],
+            "redirectAllRequestsTo must be either http or https"
+          );
+          params.WebsiteConfiguration.RedirectAllRequestsTo.Protocol = redirectAllRequestsTo.protocol;
+        }
+      } else {
+        params.WebsiteConfiguration.IndexDocument = { Suffix: indexDoc };
+        params.WebsiteConfiguration.ErrorDocument = { Key: errorDoc };
+      }
+
+      const routingRules = this.serverless.service.custom.client.routingRules;
+      if (routingRules) {
+        params.WebsiteConfiguration.RoutingRules = [];
+        
+        confirmArray(routingRules, 'routingRules must be a list');
+        
+        routingRules.forEach(r => {
+          const routingRule = {
+            Redirect: {}
+          };
+          confirmExists(r.redirect, 'redirect must be specified for each member of routingRules');
+
+          confirm(
+            !(r.redirect.replaceKeyPrefixWith && r.redirect.replaceKeyWith),
+            "replaceKeyPrefixWith and replaceKeyWith cannot both be specified"
+          );
+
+          const props = ["hostName", "httpRedirectCode", "protocol", "replaceKeyPrefixWith", "replaceKeyWith"];
+          props.forEach(p => {
+            if (r.redirect[p]) {
+              if (p === "httpRedirectCode") {
+                confirmInteger(r.redirect[p], "httpRedirectCode must be an integer");
+                r.redirect[p] = r.redirect[p].toString();
+              } else {
+                confirmString(r.redirect[p], `${p} must be a string`);
+              }
+              routingRule.Redirect[p.charAt(0).toUpperCase() + p.slice(1)] = r.redirect[p];
+            }
+          });
+
+          if (r.condition) {
+            routingRule.Condition = {};
+
+            confirm(
+              r.condition.httpErrorCodeReturnedEquals || r.condition.keyPrefixEquals,
+              "httpErrorCodeReturnedEquals or keyPrefixEquals must be defined for each condition"
+            );
+
+            const props = ["httpErrorCodeReturnedEquals", "keyPrefixEquals"];
+            props.forEach(p => {
+              if (r.condition[p]) {
+                if (p === "httpErrorCodeReturnedEquals") {
+                  confirmInteger(r.condition[p], "httpErrorCodeReturnedEquals must be an integer");
+                  r.condition[p] = r.condition[p].toString();
+                } else {
+                  confirmString(r.condition[p], `${p} must be a string`);
+                }
+                routingRule.Condition[p.charAt(0).toUpperCase() + p.slice(1)] = r.condition[p];
+              }
+            });
+          }
+
+          params.WebsiteConfiguration.RoutingRules.push(routingRule);
+        });
+        
+      }
+
+      function confirm(expr, error) {
```

Comment:
This is kinda awkward and uses the same name as the built-in browser function confirm(). 

What do you think about using something like this - https://github.com/arasatasaygin/is.js

Might still need a little custom work to implement it but seems like it might save us defining a bunch of custom checks. If we're going to define those they should be broken out of an already unwieldy index.js

Reply:
Whoops, I had completely forgotten that confirm was a built-in...I never really use it. The library you linked looks like a great option for the kind of stuff that I was rolling-my-own on. I'm happy to switch to that. Thanks!